### PR TITLE
add comment about specific method docstrings

### PR DESCRIPTION
**Pull request by Datseris**

### CODE REVIEW

File: docs/src/man/syntax.md [hunk 1]
```diff
@@ -40,6 +40,17 @@ which will cause any unlisted docstrings to raise warnings when [`makedocs`](@re
...
+Notice also that you can use `@docs` to display the documentation strings of only specific
+methods, by stating the dispatch types. For example
+````markdown
+```@docs
+f(::Type1, ::Type2)
+```
+````
+will only display the documentation string of `f` that is related to these types.
+This can be useful when your module extends a function and adds a documentation
+string to that extention.
```
Comment (mortenpi) [inline docs/src/man/syntax.md]: Thanks, @Datseris! I'll just add a small note that the signature should match exactly (related issue: #839).

```suggestion
string to that new method. Note that the method signature has to match exactly -- Documenter will
not match methods based on abstract types and dispatch rules.
```

**Comment by Datseris** [comment]
@mortenpi I have added your suggestion, but I think you need to come up with an illustrative example. The currently standing text is confusing, at least for me. The problem is, it is not clear to me what "exactly" actually means. E.g. how does one include the method `f(::Vector{T}) where {T<:MyTypes}`?

**Comment by mortenpi** [comment]
Parametric methods are actually problematic currently (#839) and I might refrain from commenting about them at the moment. But what I meant is this: suppose you have a method

```julia
foo(::Integer) = x
```

that has a docstring. Then neither `foo(::Number)` nor `foo(::Int64)` will match it in an at-docs block. The only way you can get that docstring is by listing exactly `foo(::Integer)` in the at-docs block.

**Comment by Datseris** [comment]
Okay, I see your point. Then can you please add a comment that says exactly what you just told me, for the non-parametric methids? (I can do it instead but you should take the commit credit :D )

**Comment by mortenpi** [comment]
Done. Does it look good to you @Datseris?

I will mention that it's actually completely accurate. If you only have the `foo(::Integer)` method and you put `foo(::Number)` into at-docs, it will actually splice `foo(::Integer)` into the docs. But I consider that to be a bug, since if you add a docstring to `foo(::Number)`, it will only splice the `::Number` one and not `::Integer`. So we are now documenting what I would consider to be the correct behavior and need to fix Documenter at some point.

> I can do it instead but you should take the commit credit :D

Gonna squash anyway, so not particularly worried about it :smile:

**Comment by Datseris** [comment]
Looks good to me! If you agree then you should merge! :)

**Pull request merged**